app/src/main/java/com/example/bluetoothdevicesapplication/ui/view/DevicesRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,55 @@
+package com.example.bluetoothdevicesapplication.ui.view
+
+import android.bluetooth.BluetoothDevice
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.RecyclerView
+import com.example.bluetoothdevicesapplication.R
+import kotlinx.android.synthetic.main.list_item.view.*
+import java.lang.reflect.Method
+
+class DevicesRecyclerViewAdapter(private val pairedDevices: List<BluetoothDevice>) :
+    RecyclerView.Adapter<DevicesRecyclerViewAdapter.DevicesRecyclerViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DevicesRecyclerViewHolder {
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.list_item ,parent, false)
+
+        return DevicesRecyclerViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: DevicesRecyclerViewHolder, position: Int) {
+        val currentDevice = pairedDevices[position]
+        holder.textView.text = currentDevice.name
+        val m: Method = currentDevice.javaClass.getMethod("isConnected")
```

Comment:
Is there really no other way do do this than reflection? Well done finding this, Maybe this helps:

https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/bluetooth/BluetoothManager.java;l=140?q=BluetoothManager

Reply:
This sadly didn't work for me, it always returned an empty list. I did some research and it seems that it only works for GATT devices (has something to do with Bluetooth low energy devices) [https://stackoverflow.com/a/45327459](https://stackoverflow.com/a/45327459). 
I got this solution from [https://stackoverflow.com/a/58882930](https://stackoverflow.com/a/58882930) as nothing else worked for me but maybe I am missing something.

Reply:
I see, yeah might be that we are using internal AOSP APIs for that